python/pyarrow/tests/test_extension_type.py

```diff
@@ -445,22 +445,28 @@ def test_parquet(tmpdir, registered_period_type):
     import base64
     decoded_schema = base64.b64decode(meta.metadata[b"ARROW:schema"])
     schema = pa.ipc.read_schema(pa.BufferReader(decoded_schema))
-    assert schema.field("ext").metadata == {
-        b'ARROW:extension:metadata': b'freq=D',
-        b'ARROW:extension:name': b'pandas.period'}
+    # Since the type could be reconstructed, the extension type metadata is
+    # absent.
+    assert schema.field("ext").metadata == {}
```

Comment:
@jorisvandenbossche You may want to check whether this is ok for you.

Reply:
I don't have a fully good sense of what people do with this metadata, but I suppose when the type was recognized, it should not be a problem if the metadata is not present anymore (since you can always retrieve it again from the type instance).

I know Micah mentioned they were using those metadata in BigQuery, but without a registered extension type, so such use case should not be impacted.

This test is for parquet roundtrip, but a similar change was done for plain IPC roundtrip as well? (that currently also preserves the field metadata, even when the type was recognized)

Reply:
The change was actually done on the IPC side and Parquet inherited it, yes.